src/interpreter/executors/main.rs

```diff
@@ -226,6 +113,153 @@ where
             }
         }
     }
+}
+
+impl<S, Tx> Interpreter<S, Tx>
+where
+    S: InterpreterStorage,
+{
+    fn _deploy(&mut self, mut create: Create, initial_balances: InitialBalances) -> Result<Create, InterpreterError> {
+        let salt = create.salt();
+        let storage_slots = create.storage_slots();
+        let contract = Contract::try_from(&create)?;
+        let root = contract.root();
+        let storage_root = Contract::initial_state_root(storage_slots.iter());
+        let id = contract.id(salt, &root, &storage_root);
+
+        if !create
+            .outputs()
+            .iter()
+            .any(|output| matches!(output, Output::ContractCreated { contract_id, state_root } if contract_id == &id && state_root == &storage_root))
+        {
+            return Err(InterpreterError::Panic(PanicReason::ContractNotInInputs));
+        }
+
+        let storage = &mut self.storage;
+        storage
+            .deploy_contract_with_id(salt, storage_slots, &contract, &root, &id)
+            .map_err(InterpreterError::from_io)?;
+
+        // TODO: Maybe the code below is useless because we can't spend any gas and can't execute
+        //  any code, so the output also should be the same.
+        let remaining_gas = create.limit();
+        Self::finalize_outputs(
+            &mut create,
+            false,
+            remaining_gas,
+            &initial_balances,
+            &RuntimeBalances::from(initial_balances.clone()),
+            &self.params,
+        )?;
+        Ok(create)
+    }
+}
+
+impl<S, Tx> Interpreter<S, Tx>
+where
+    S: InterpreterStorage,
+    Tx: ExecutableTransaction,
+{
+    pub(crate) fn run(&mut self) -> Result<ProgramState, InterpreterError> {
+        // TODO: Remove `Create` from here
```

Comment:
Before merging the PR, I will create according issue with the description

The main idea is that the `transact` method in the future should work only with the `Script` transaction. For `Create` there exists a `deploy` method that should be used.